src/npc.h

```diff
@@ -877,6 +877,11 @@ class npc : public player
         bool wants_to_buy( const item &it ) const;
         bool wants_to_buy( const item &/*it*/, int at_price, int /*market_price*/ ) const;
 
+        // Will swap items with player at no charge. (Eg: NPC ally)
```

Comment:
Please don't put documentation for functions in the header **and** in the cpp file. This is very confusing. A reader will typically expected the documentation at either place, but not at both. A future coder may also easily forget to update the documentation at *both* places.